src/components/themed/SimpleTextInput.tsx

```diff
@@ -179,52 +184,61 @@ export const SimpleTextInput = React.forwardRef<SimpleTextInputRef, SimpleTextIn
-            onSubmitEditing={handleSubmitEditing}
-            maxLength={maxLength}
-            // Other Props:
-            autoCapitalize={autoCapitalize}
-            autoCorrect={autoCorrect}
-            blurOnSubmit={blurOnSubmit}
-            inputAccessoryViewID={inputAccessoryViewID}
-            secureTextEntry={secureTextEntry}
-          />
-        </InnerContainer>
-
-        <TouchContainer accessible onPress={handleClearPress} testID={`${testID}.clearIcon`}>
-          <SideContainer size={rightIconSize}>
-            <CloseIconAnimated color={iconColor} size={rightIconSize} />
-          </SideContainer>
-        </TouchContainer>
-      </Container>
-    </EdgeTouchableWithoutFeedback>
+    <ContainerView>
+      <EdgeTouchableWithoutFeedback accessible={false} testID={testID} onPress={() => focus()}>
+        <InputContainerView disableAnimation={disableAnimation} focusAnimation={focusAnimation} scale={scale} spaceProps={spaceProps}>
+          <SideContainer size={leftIconSize}>{Icon == null ? null : <Icon color={iconColor} size={leftIconSize} />}</SideContainer>
+
+          <InnerContainer>
+            <InputField
+              accessible
+              ref={inputRef}
+              keyboardType={props.keyboardType}
+              returnKeyType={props.returnKeyType}
+              accessibilityState={{ disabled }}
+              autoFocus={autoFocus}
+              disableAnimation={disableAnimation}
+              focusAnimation={focusAnimation}
+              placeholder={placeholder}
+              placeholderTextColor={placeholderTextColor}
+              selectionColor={theme.textInputTextColor}
+              testID={`${testID}.textInput`}
+              textAlignVertical="top"
+              value={value}
+              // Callbacks:
+              onBlur={handleBlur}
+              onChangeText={onChangeText}
+              onFocus={handleFocus}
+              onSubmitEditing={handleSubmitEditing}
+              maxLength={maxLength}
+              // Other Props:
+              autoCapitalize={autoCapitalize}
+              autoCorrect={autoCorrect}
+              blurOnSubmit={blurOnSubmit}
+              inputAccessoryViewID={inputAccessoryViewID}
+              secureTextEntry={secureTextEntry}
+            />
+          </InnerContainer>
+
+          <TouchContainer accessible onPress={handleClearPress} testID={`${testID}.clearIcon`}>
+            <SideContainer size={rightIconSize}>
+              <CloseIconAnimated color={iconColor} size={rightIconSize} />
+            </SideContainer>
+          </TouchContainer>
+        </InputContainerView>
+      </EdgeTouchableWithoutFeedback>
+      {isFocused ? (
+        <ButtonUi4 onPress={handleDonePress} type="tertiary" label={lstrings.string_done_cap} testID={`${testID}.doneButton`} marginRem={[0.5, 0, 0.5, 0.5]} />
+      ) : null}
+    </ContainerView>
   )
 })
 
-const Container = styled(Animated.View)<{
+const ContainerView = styled(View)({
```

Comment:
`Container` is kind of redundant, the `View` already carries that meaning. Would be more descriptive to use `RowView,` then we can reuse it as an export later

Reply:
These styled components are private and we should never export them. Container is a better name then row because this isn't a "row", it's the container for the all of the parts of the SimpleTextInput component. A row implies that it is a sibling of other rows.

Reply:
The word "row" on its own has no implications towards parent/child relationships. It simply describes a row layout as opposed to a column layout.

One of my gripes with this pattern is that important information is abstracted away from the JSX. In the traditional `styles.X` pattern, often times simple styles such as `flexDirection: "row"` are named as such, so that it is easy to grok at the layout of the component without having to jump to a definition.

Reply:
Why name the component based on it's implementation details? We never required naming styles based on their properties before. Just because the container implements a row layout, doesn't mean it _is_ a row. Semantics for this is that it is a container. If it had a blue background, would we have to name it blue? If we change the theme colors, from blue to pink, do we need to change the component name? Same goes for the flex direction; if we change from row to column later perhaps if we need to add more elements and change the implementation, then now we might accidentally not rename the component and then it'd be confusing. 

I say we shouldn't ever name components based on the implementation details. I've been naming things by some higher-order semantic. In this case, a container is a component which is responsible for containing other components and that's it's purpose. It's similar in spirit to the name "wrapper", but I just preferred "container" as the word to use.